miniapps/navier/navier_mms.cpp

```diff
@@ -0,0 +1,237 @@
+             * sin(2.0 * M_PI * xi) * sin(M_PI * yi))
+          + M_PI
+          * (2.0 * ctx.kinvis * pow(M_PI, 2.0)
+             * (1.0 - 2.0 * cos(2.0 * M_PI * xi)) * sin(t)
+             + cos(t) * pow(sin(M_PI * xi), 2.0))
+          * sin(2.0 * M_PI * yi);
+
+   u(1) = M_PI * cos(M_PI * yi) * sin(t)
+          * (cos(M_PI * xi)
+             + 2.0 * ctx.kinvis * pow(M_PI, 2.0) * cos(M_PI * yi)
+             * sin(2.0 * M_PI * xi))
+          - M_PI * (cos(t) + 6.0 * ctx.kinvis * pow(M_PI, 2.0) * sin(t))
+          * sin(2.0 * M_PI * xi) * pow(sin(M_PI * yi), 2.0)
+          + 4.0 * pow(M_PI, 3.0) * cos(M_PI * yi) * pow(sin(t), 2.0)
+          * pow(sin(M_PI * xi), 2.0) * pow(sin(M_PI * yi), 3.0);
+}
+
+int main(int argc, char *argv[])
+{
+   MPI_Session mpi(argc, argv);
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&ctx.ser_ref_levels,
+                  "-rs",
+                  "--refine-serial",
+                  "Number of times to refine the mesh uniformly in serial.");
+   args.AddOption(&ctx.order,
+                  "-o",
+                  "--order",
+                  "Order (degree) of the finite elements.");
+   args.AddOption(&ctx.dt, "-dt", "--time-step", "Time step.");
+   args.AddOption(&ctx.t_final, "-tf", "--final-time", "Final time.");
+   args.AddOption(&ctx.pa,
+                  "-pa",
+                  "--enable-pa",
+                  "-no-pa",
+                  "--disable-pa",
+                  "Enable partial assembly.");
+   args.AddOption(&ctx.ni,
+                  "-ni",
+                  "--enable-ni",
+                  "-no-ni",
+                  "--disable-ni",
+                  "Enable numerical integration rules.");
+   args.AddOption(&ctx.visualization,
+                  "-vis",
+                  "--visualization",
+                  "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.AddOption(
+      &ctx.checkres,
+      "-cr",
+      "--checkresult",
+      "-no-cr",
+      "--no-checkresult",
+      "Enable or disable checking of the result. Returns -1 on failure.");
+   args.Parse();
+   if (!args.Good())
+   {
+      if (mpi.Root())
+      {
+         args.PrintUsage(mfem::out);
+      }
+      MPI_Finalize();
+      return 1;
+   }
+   if (mpi.Root())
+   {
+      args.PrintOptions(mfem::out);
+   }
+
+   Mesh *mesh = new Mesh("../../data/inline-quad.mesh");
+   mesh->EnsureNodes();
+   GridFunction *nodes = mesh->GetNodes();
+   *nodes *= 2.0;
+   *nodes -= 1.0;
+
+   for (int i = 0; i < ctx.ser_ref_levels; ++i)
+   {
+      mesh->UniformRefinement();
+   }
+
+   if (mpi.Root())
+   {
+      std::cout << "Number of elements: " << mesh->GetNE() << std::endl;
+   }
+
+   ParMesh *pmesh = new ParMesh(MPI_COMM_WORLD, *mesh);
+   delete mesh;
+
+   // Create the flow solver.
+   NavierSolver naviersolver(pmesh, ctx.order, ctx.kinvis);
+   naviersolver.EnablePA(true);
```

Comment:
This line (and the next line) should be fixed in all miniapps so that it takes input based on the command line option.